llama_stack/distribution/server/quota.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import json
+from datetime import datetime, timezone
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from llama_stack.log import get_logger
+from llama_stack.providers.utils.kvstore.api import KVStore
+from llama_stack.providers.utils.kvstore.config import KVStoreConfig, SqliteKVStoreConfig
+from llama_stack.providers.utils.kvstore.kvstore import kvstore_impl
+
+logger = get_logger(name=__name__, category="quota")
+
+
+class QuotaMiddleware:
+    """
+    ASGI middleware enforcing per client daily request quotas.
+
+    Expects Authorization: Bearer <client_id> header.
+    Tracks counts in a KV store (SQLite); returns HTTP 429 when limit is exceeded.
+    """
+
+    def __init__(
+        self,
+        app: ASGIApp,
+        kv_config: KVStoreConfig | None = None,
+        default_requests_per_day: int = 1000,
+        window_seconds: int = 86400,
+    ):
+        self.app = app
+        # if no config passed, default to on disk SQLite
+        self._kv_config = kv_config or SqliteKVStoreConfig(db_path="./quotas.db")
+        self._kv: KVStore | None = None
+        self.default_limit = default_requests_per_day
+        self.window = window_seconds
+
+    async def _get_kv(self) -> KVStore:
+        if self._kv is None:
+            self._kv = await kvstore_impl(self._kv_config)
+        return self._kv
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send):
+        if scope["type"] == "http":
+            client_id = scope.get("authenticated_client_id")
+            if not client_id:
+                logger.error(
+                    "QuotaMiddleware requires an authenticated client_id but none was found in the scope. "
+                    "This likely means AuthenticationMiddleware is not installed or failed."
+                )
+                return await self._send_error(
+                    send, 500, "Quota system misconfigured: missing authenticated client identity"
+                )
+
+            key = f"quota:{client_id}:{datetime.now(timezone.utc).date().isoformat()}"
+
+            try:
+                kv = await self._get_kv()
+                prev = await kv.get(key) or "0"
+                count = int(prev) + 1
+                await kv.set(key, str(count))
+                # Note: TTL/expire is only supported on backends that implement it;
+                # for SQLite we ignore expire.
+            except Exception:
+                logger.exception("Error accessing KV store for quota")
```

Comment:
```suggestion
                logger.exception("failed to access KV store for quota")
```

we use this convention almost across the codebase for errors :)